runtime/ts/keymgmt/webcrypto.ts

```diff
@@ -0,0 +1,276 @@
+import {DeviceKey, Key, PrivateKey, PublicKey, RecoveryKey, SessionKey, WrappedKey} from "./keys";
+import {KeyGenerator, KeyStorage} from "./manager";
+import idb, {ObjectStore} from 'idb';
+import {decode, encode} from './base64';
+import rs from 'jsrsasign';
+
+class WebCryptoStorableKey<T> {
+
+    protected key: T;
+
+    constructor(key: T) {
+        this.key = key;
+    }
+
+    algorithm(): string {
+        // @ts-ignore
+        return (this.key as CryptoKey).algorithm.name;
+    }
+
+    storableKey(): T {
+        return this.key;
+    }
+}
+
+class WebCryptoWrappedKey implements WrappedKey {
+    private wrappedKeyData: Uint8Array;
+    private wrappedBy: PublicKey;
+
+    constructor(wrappedKeyData: Uint8Array, wrappedBy: PublicKey) {
+        this.wrappedKeyData = wrappedKeyData;
+        this.wrappedBy = wrappedBy;
+    }
+
+    algorithm(): string {
+        return this.wrappedBy.algorithm();
+    }
+
+    private unwrap(privKey: PrivateKey): PromiseLike<SessionKey> {
+        const webPrivKey = privKey as WebCryptoPrivateKey;
+
+        return crypto.subtle.unwrapKey(
```

Comment:
Add a comment here briefly explaining why we don't just generate random numbers for the shared secret.